zenoh/src/net/routing/hat/p2p_peer/interests.rs

```diff
@@ -147,7 +147,8 @@ impl HatInterestTrait for HatCode {
         };
         for dst_face in tables.faces.values_mut().filter(|f| {
             f.whatami == WhatAmI::Router
-                || (options.tokens()
+                || (face.whatami == WhatAmI::Client
```

Comment:
I suggest writing this as:

```rust
match dst_face.whatami {
    WhatAmI::Router => true,
    WhatAmI::Client => false,
    WhatAmI::Peer => options.tokens() 
        && src_face.whatami == WhatAmI::Client
        && mode == InterestMode::Current 
        && !initial_interest(f).map(|i| i.finalized).unwrap_or(true))
}
```

So that the condition reads more like a case by case analysis.

Reply:
Just peaking over here but maybe:
```rust
WhatAmI::Peer => {
        options.tokens() 
            && src_face.whatami == WhatAmI::Client
            && mode == InterestMode::Current
            && !initial_interest(f).map_or(true, |i| i.finalized)
    }
```
Instead?